Collections/script.js

```diff
@@ -256,14 +256,137 @@ console.log(filterByAge(people, 30));
+let day = birthday.toLocaleDateString("en", {weekday: "long"}); //musiałam wygooglować i poszukać więcej o obiekcie Date
+console.log("I was born on", day += ".");
+
 // Zadanie 15.
 
+//to zadanie jest dla mnie zbyt skomplikowane i mimo ze CHAT GPT wygenerował mi rozwiązanie i komentarze, to jest to na tyle trudne, ze nie rozumiem nawet wyjaśnień, więc myślę, ze to zadanie odpuszczę w pracy domowej- kompletnie nie rozumiem.
+
 // Zadanie 16.
+//okej wydawało się łatwe, totalnie level hard jak dla mnie- nie wiedziałam za bardzo od czego zacząć, więc zaczęłam coś skrobać tylko dla let a, zeby jakkolwiek to pojąć, a dopiero później edytowałam całą funkcję 
+
+let a = "java_script";
+let b = "java-script";
+let c = "Java-Script";
+let d = "asp_Net_Core";
+
+function toCamelCase (input) {
+    let parts = input.split(/-|_/); //ok tutaj definiujemy sobie rozdzielanie stringu za pomocą - lub _
+
+    let camelCaseParts = parts.map(function(part, index){
+    if(index === 0) {
+        return part;
+    } else {
+        return part[0].toUpperCase() + part.slice(1);
+    }}) //tutaj warunkujemy, ze jeśli indeks części stringu wynosi 0 (np. w java-script mamy dwie części, java i script, więc mowa tutaj o części 'java'), to zwrócone ma zostać oryginalne słowo, a w innych przypadkach (czyli np. 'script') mamy zmienić pierwszę literę stringu danej części na wielką i zwrócić resztę tej części bez pierwszej litery, która była mała (sama bym na to nie wpadła, ale zwrocenie part z podmieniona tylko wielką literą jest jeszcz gorsze)
+
+    let camelCase = camelCaseParts.join(''); //definiujemy sobie finalną zmienną, no bo trzeba to wszystko ze sobą połączyć
+    return camelCase;
+};
+
+
+console.log(toCamelCase(a));
+console.log(toCamelCase(b));
+console.log(toCamelCase(c));
+console.log(toCamelCase(d));
+
 
 // Zadanie 17.
+
+let studentsGrades =([
+	{
+		id: 1,
+		name: "Jacek",
+		grades: [5, 3, 4, 2, 5, 5],
+	},
+	{
+		id: 2,
+		name: "Ewa",
+		grades: [2, 3, 3, 3, 2, 5],
+	},
+	{
+		id: 3,
+		name: "Zygmunt",
+		grades: [2, 2, 4, 4, 3, 3],
+	},
+]);
+
+let topGrades = studentsGrades.map(function(student){ //ok zadeklarowałam sobie nową zmienną topGrades, która da mi nową tablicę po uzyciu funkcji map, która przechodzi przez wszystkie elementy tablicy i zwraca nową (nazwałam kazdy element tablicy 'student')
+    return student.grades.reduce(function(currentMax, gradeIndex){ 
+        return Math.max(currentMax, gradeIndex);}, -Infinity); //z tym mam problem (zadna nowość XD), bo nie mogę zapamiętać, ze ta pierwsza wartość, to aktualna wartość, jaką mamy na danym indeksie, która jest równiez początkową wartością (tutaj -infinity) i później zmienia się zgodnie z indeksem i jeśli jest większa (w tym przypadku, bo mamy math.max), to jest zastępowana lizbą z kolejnego indeksu, niby ma to sens jak się o tym czyta, ale sama bym pewnie na to nie wpadła
+    });
```

Comment:
Powyższa funkcja nie radzi sobie z tą częścią zadania:
`Jeśli uczeń nie ma ocen, to przyjmijmy, że jego najwyższa ocena jest równa 0.`